src/main/java/net/boomerangplatform/service/LogServiceImpl.java

```diff
@@ -0,0 +1,185 @@
+package net.boomerangplatform.service;
+
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.search.ClearScrollRequest;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.SearchScrollRequest;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.MatchPhraseQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.Scroll;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.MessageSource;
+import org.springframework.context.support.MessageSourceAccessor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+
+import net.boomerangplatform.error.BoomerangException;
+import net.boomerangplatform.kube.exception.KubeRuntimeException;
+import net.boomerangplatform.kube.service.AbstractKubeServiceImpl;
+
+public class LogServiceImpl implements LogService {
+
+	  private static final Logger LOGGER = LogManager.getLogger(LogServiceImpl.class);
+
+	  @Value("${kube.worker.logging.type}")
+	  protected String loggingType;
+
+	  @Autowired
+	  private MessageSource messageSource;
+
+	  @Autowired(required = false)
+	  private RestHighLevelClient elasticRestClient;
+	  
+	  @Autowired
+	  private AbstractKubeServiceImpl kubeService;
+
+//	  TODO: Confirm if this is even used anywhere...
+//	  @Override
+//	  public Response getLogForTask(String workflowId, String workflowActivityId, String taskId,  String taskActivityId) {
+//	    Response response = new Response("0", "");
+//	    try {
+//	        if (kubeService.isKubePodAvailable(workflowId, workflowActivityId, taskId) && !streamLogsFromElastic() && !streamLogsFromLoki()) {
+//	        	response.setMessage(kubeService.getPodLog(workflowId, workflowActivityId, taskId, taskActivityId));
+//	            
+//	          } else if (streamLogsFromElastic()) {
+//	        	  return streamLogsFromElastic(workflowActivityId);
+//	          } else if (streamLogsFromLoki()) {
+//	        	  //TODO Loki Implementation
+//	          } else {
+//	        	  //TODO throw error that there is no log
+//	          }
+//	    } catch (KubeRuntimeException e) {
+//			  throw new BoomerangException(e, 1, e.toString(), HttpStatus.INTERNAL_SERVER_ERROR);
+//	    }
+//	    return response;
+//	  }
```

Comment:
@marcusdroy this is the part that I need confirmation on whether we need to develop out, otherwise we just have a stream only endpoint